actionview/lib/action_view/template.rb

```diff
@@ -293,7 +294,7 @@ def inspect
     end
 
     def source
-      @source.to_s
+      @source ||= @_source.to_s.dup
```

Comment:
```suggestion
      @source_string ||= @source.to_s
```

I'm not sure I see the benefit of `dup` here. 🤔 Previously we exposed the return value of `to_s` from the source (so if some `@source` implementation cached the value it would already be mutable), and since Template is now memoizing the value, the `dup` isn't protecting the Template's value from being mutated either.